tools/engine_tool/lib/src/commands/lint_command.dart

```diff
@@ -0,0 +1,82 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io' show File;
+
+import 'package:path/path.dart' as p;
+
+import '../dart_utils.dart';
+import '../proc_utils.dart';
+import 'command.dart';
+import 'flags.dart';
+
+/// The root 'lint' command.
+final class LintCommand extends CommandBase {
+  /// Constructs the 'lint' command.
+  LintCommand({
+    required super.environment,
+  }) {
+    _linterArgs['dart'] = <String>[findDartBinDirectory(environment)];
+    _linterRunner['java'] = p.join(findDartBinDirectory(environment), 'dart');
+  }
+
+  // Map from lint-type to path of tool to invoke.
+  // Paths are relative to 'engine/src/flutter'.
+  static const Map<String, String> _linters = <String, String>{
+    'dart': 'ci/analyze.sh',
```

Comment:
Given we're referring to these keys in multiple places, I'd just make this a first class enum:

```dart
enum Language {
  dart,
  python,
  c,
  java,
}
```

You could even include more information if you wanted (like the paths, args, runners) instead of having multiple maps.

Reply:
I've gone with this but instead of `Language` I went with `Linter` as we will have non-language specific lints.

Reply:
I've also consolidated all the maps down to one.